### ISSUE TITLE
Rule: Function expressions must have names

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#595]
A pattern that's becoming more common is to give function expressions names to aid in debugging, such as:

``` js
Foo.prototype.bar = function bar(){}
```

This rule should warn when a function expression is missing a name.

The rule ID should be `func-names` and should be off by default.
